src/mapping/RadialGeoMultiscaleMapping.cpp

```diff
@@ -0,0 +1,218 @@
+      Eigen::VectorXd axisMidpoints(outSize);
+      auto &          outputVerticesRef = output()->vertices();
+
+      // Order the vertices of the 1D output mesh, to correctly compute the midpoints of the segments between neighboring vertices
+      std::vector<size_t> ordered_vertex_indices(output()->vertices().size());
+      std::iota(ordered_vertex_indices.begin(), ordered_vertex_indices.end(), 0);
+      std::sort(ordered_vertex_indices.begin(), ordered_vertex_indices.end(),
+                [effectiveCoordinate, &outputVerticesRef](const size_t aindex, const size_t bindex) {
+                  return outputVerticesRef[aindex].getCoords()[effectiveCoordinate] < outputVerticesRef[bindex].getCoords()[effectiveCoordinate];
+                });
+      // Compute the midpoints of the 1D output mesh
+      for (size_t i = 0; i < (outSize - 1); i++) {
+        auto axisPositionCurrent = output()->vertices()[i].getCoords()[effectiveCoordinate];
+        auto axisPositionNext    = output()->vertices()[i + 1].getCoords()[effectiveCoordinate];
+        axisMidpoints(i)         = (axisPositionCurrent + axisPositionNext) / 2;
+      }
+      axisMidpoints(outSize - 1) = std::numeric_limits<double>::max(); // large number, such that vertices after the last midpoint are still assigned
+
+      Eigen::VectorXi counters(outSize); // counts number of vertices in between midpoints for averaging
+      counters = Eigen::VectorXi::Zero(outSize);
+
+      // Identify which vertex (index) of the 3D mesh corresponds to which vertex (index) of the 1D meesh
+      _vertexIndicesCollect.clear();
+      _vertexIndicesCollect.reserve(input()->vertices().size());
+      for (size_t i = 0; i < inSize; i++) {
+        auto vertexCoords = input()->vertices()[i].getCoords()[effectiveCoordinate];
+        int  index        = 0;
+        while (vertexCoords > axisMidpoints(index)) {
+          index++;
+          PRECICE_ASSERT(index < static_cast<int>(outSize));
+        }
+        _vertexIndicesCollect.push_back(index);
+        counters(index) += 1;
+      }
+      _vertexCounter = std::move(counters);
+    }
+
+  } else {
+    PRECICE_ASSERT(getConstraint() == CONSERVATIVE);
+    PRECICE_ASSERT(false, "Not yet implemented");
+    PRECICE_DEBUG("Compute conservative mapping");
+  }
+  _hasComputedMapping = true;
+}
+
+void RadialGeoMultiscaleMapping::clear()
+{
+  PRECICE_TRACE();
+  _vertexIndicesSpread.clear();
+  _vertexIndicesCollect.clear();
+  _hasComputedMapping = false;
+}
+
+void RadialGeoMultiscaleMapping::mapConservative(const time::Sample &inData, Eigen::VectorXd &outData)
+{
+  PRECICE_ASSERT(getConstraint() == CONSERVATIVE);
+  PRECICE_ASSERT(false, "Not yet implemented");
+  PRECICE_DEBUG("Map conservative");
+}
+
+void RadialGeoMultiscaleMapping::mapConsistent(const time::Sample &inData, Eigen::VectorXd &outData)
+{
+  PRECICE_TRACE();
+
+  const int              inDataDimensions = inData.dataDims;
```

Comment:
After converting to samples (starting from https://github.com/precice/precice/pull/1675/commits/0f0286e8169c5cddeb5b8a52424b58ffa430f278 (axial) / https://github.com/precice/precice/pull/1675/commits/abab8f41aa97db2b4cf644a43b9682b14dbd3f28 (radial)), the `inData.dataDims` return `1` instead of `3`, **only in some platforms**, which sounds like a bug.

I cannot trace where exactly this comes from, but I only seem to be explicitly defining 3D samples.

@fsimonis any clue what is going on here?

Reply:
`testConsistentSpreadX` results in the following meshes after mapping:

```
nValues =   1 0 0
            2 0 0
            3 0 0

outValues = 1 0 0
            1 0 0
            0 0 0
            0 0 0
            0 0 0
            0 0 0
```

The first two points are correct, but it looks like the mapping did nothing for the rest of the points.

I get the _exact same results_ for `testConsistentSpreadZ`, which makes me wonder whether the X or Z makes any difference in the test or in the mapping.

Reply:
The reason that only the `Bare` CI targets pass is because they only execute smoke tests and not the regular unit and integration tests. @fsimonis is that intentional, or a blind spot?

Reply:
After fixing the `_vertexIndicesSpread` issue, I get the following results:

```
nValues =   1 0 0
            2 0 0
            3 0 0

outValues = 1 0 0
            1 0 0
            2 0 0
            2 0 0
            3 0 0
            3 0 0
```

which are the intended, for both spread-x and spread-z, but thee tests still fail for the collect:

```
inValues =   1 0 0
             3 0 0
             5 0 0
             7 0 0
             2 0 0
             4 0 0

outValues =  2 0 0
            12 0 0
             6 0 0
```

again, I get exactly the same results for collect-X and collect-Y. It looks like the first two points were summed and divided by 2, but the rest were only (correctly) summed, which points me to a similar confusion in the respective loops in terms of their indices and bounds.

Reply:
Fixed those as well, similar issue: I had to convert from vertex index to data index.

Conclusion 1: Using a 3D mesh to represent 1D data is very error-prone. We desperately need proper support for 1D data.

Conclusion 2: The tests do the right thing! And they should now all pass.